packages/react-hooks/src/useSubscription/useSubscription.ts

```diff
@@ -0,0 +1,55 @@
+import { MedplumClient, SubscriptionEmitter, SubscriptionEventMap } from '@medplum/core';
+import { Bundle } from '@medplum/fhirtypes';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useMedplum } from '../MedplumProvider/MedplumProvider.context';
+
+const SUBSCRIPTION_DEBOUNCE_MS = 3000;
+
+export function useSubscription(criteria: string, callback: (bundle: Bundle) => void): void {
```

Comment:
This is getting kinda heavy with 3 `useEffect`... Feels like we should be able to slim this down.  In my mind, it should be possible for this to be a very lightweight operation, so that we can freely call `useSubscription` without worrying about performance overhead.

Reply:
Talked about this offline:

Good call raising concern about multiple `useEffect`, they add up quickly. We determined that these should be pretty light and shouldn't rerun except when logging out or if the criteria changes... 

I did reduce the amount of effects by 33% (3 -> 2) so that should help a little 👍